### DISCUSSION TITLE
v7.3.0 Release of PowerShell

### DISCUSSION THREAD

**Discussion by adityapatwardhan** [PowerShell/PowerShell#18494] [General]
### General Cmdlet Updates and Fixes

- Correct calling cmdlet `New-PSSessionOption` in script for `Restart-Computer` (#18374)

### Tests

- Add test for framework dependent package in release pipeline (Internal 23139)

### Build and Packaging Improvements

<details>

<summary>

<p>Bump to use internal .NET 7 GA build (Internal 23096)</p>

</summary>

<ul>
<li>Fix issues with building test artifacts (Internal 23116)</li>
<li>Use <code>AzFileCopy</code> task instead of <code>AzCopy.exe</code></li>
<li>Remove <code>AzCopy</code> installation from msixbundle step</li>
<li>Add TSAUpload for APIScan (#18446)</li>
<li>Add authenticode signing for assemblies on Linux builds (#18440)</li>
<li>Do not remove <code>penimc_cor3.dll</code> from build (#18438)</li>
<li>Allow two-digit revisions in vPack package validation pattern (#18392)</li>
<li>Bump <code>Microsoft.PowerShell.Native</code> from <code>7.3.0-rc.1</code> to <code>7.3.0</code> (#18413)</li>
</ul>

</details>

[7.3.0]: https://github.com/PowerShell/PowerShell/compare/v7.3.0-rc.1...v7.3.0

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - 2748D1008F025B4BBF35B326F8113E2A0DD7C3F096F669FDACC04EF1171F1C4A
- powershell_7.3.0-1.deb_amd64.deb
  - A99E9FE370B0D9C05C736FA81521D32375DDB418AB75C76D9D0A14B4CE3D3DF2
- powershell-7.3.0-1.cm.x86_64.rpm
  - CFB97B059C08F0D31EABFE854048D2ABD67A56175A896A63E7B8FB17E659EE51
- powershell-7.3.0-1.rh.x86_64.rpm
  - 3C4B9BEB52CC4F52C52DFBA59346E23DF38CFF9FE18D534F83CB902472D28B37
- powershell-7.3.0-linux-alpine-x64.tar.gz
  - 0E91269D2AA29C9D28CC921562B41514C757C9FEE9E93098F69CCB01E5E156D1
- powershell-7.3.0-linux-arm32.tar.gz
  - 5246CEC2ABA67CF3A1B7DC61FD015D35E651A1EC9D5AAA4B6A19F9B34F671616
- powershell-7.3.0-linux-arm64.tar.gz
  - DC84E26DF576EF8CC353850EEB5F6659F322FF999913442F472B158F943D109F
- powershell-7.3.0-linux-x64.tar.gz
  - 5F781392597776519BB3726B2D9854061017F4CC22F9F1CDAA9F26636B936CC0
- powershell-7.3.0-linux-x64-fxdependent.tar.gz
  - 485F4BB9626E0BF869F76CC77751ED037A5DB570EFABC064FF26B36E60916492
- powershell-7.3.0-osx-arm64.pkg
  - D4DD81CD9078C8B5F3C6407A7AB31FE8D4DD142926356E5DCAB6EDCAFAE83692
- powershell-7.3.0-osx-arm64.tar.gz
  - 52F7CCA7A99F5FADFD39B9F937AC09E8FAC4D39631F35D1DB10DDFC711E38F2B
- powershell-7.3.0-osx-x64.pkg
  - FC95BE4E3D99BF940B593B53A54FE7AA11B05D8815820CC162A4B04CC8811EDE
- powershell-7.3.0-osx-x64.tar.gz
  - B3205A8EDE7A90E6F4BF856AB38293680C3D03A69EE5E03935A61FEC455C0485
- PowerShell-7.3.0-win-arm32.zip
  - 90DBF3DF9E1F1F5EF3B5CBB13270887154E86A9CEFED54B1588C3EB6E223E8D3
- PowerShell-7.3.0-win-arm64.zip
  - 564C56AA84116FB3BF058D9BCB83F9D80D7DB802988E9DB5DFE82A7C4730B17D
- PowerShell-7.3.0-win-fxdependent.zip
  - 7CF427D9D773859248CC73B85572E0830113B94BF6522218AF00661A18C0E014
- PowerShell-7.3.0-win-fxdependentWinDesktop.zip
  - CE61E12FDE3CD04B581D966E765AD23F5FED75BBC24EE928186A3E47BEB9A538
- PowerShell-7.3.0-win-x64.msi
  - 5DBBFA10FD37A032414F36B9D6A8DE99E042081075DAB320C10F1F67B93E2C4E
- PowerShell-7.3.0-win-x64.zip
  - B4F0089E44E8E66975BE3D9968F320CD540D46F219415F3EC0C525BC1BF35974
- PowerShell-7.3.0-win-x86.msi
  - 9E819F88D3CC1575EF19CC1ACF002E7D942737B9D34E8A5306410B0DF34F433C
- PowerShell-7.3.0-win-x86.zip
  - 03210C0C2D29B4B4F103D7A1F6B63743A311CE87A7E75CF67DEA9EA4266EF751

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.3.0'>v7.3.0 Release of PowerShell</a>.</em>

**Comment by gutenfries** [comment]
## Release [7.3.0](https://github.com/PowerShell/PowerShell/releases/tag/v7.3.0) shows some MAJOR improvements in startup time.

On my windows x64 dev machine (normally quite fast), release [v7.2.x](https://github.com/PowerShell/PowerShell/releases/tag/v7.2.7) took upwards of 2000-3000ms at startup, just to get to the first paint. (no screenshot available sorry).

Beta-release [v7.3.0-rc.1](https://github.com/PowerShell/PowerShell/releases/tag/v7.3.0-rc.1) ran significantly faster than the previous version ([v7.2.7 stable](https://github.com/PowerShell/PowerShell/releases/tag/v7.2.7)), and it great to see that this finally made it to stable for PS7

### At the end of the day, you can expect <1000ms startup times (tested on x64 windows), depending on your startup scripts (I have a generous amount running at startup)

---
![fast start-up time of PowerShell 7](https://user-images.githubusercontent.com/79616833/201352228-7097315c-6d87-4527-a50e-728ae01106a1.png)

**Comment by krish2kdev** [comment]
Can anyone help me which command to upgrade the powershell?

**Reply by iSazonov** [reply]
Windows or *nix? How did you install pwsh early?

**Comment by sergeevabc** [comment]


**Reply by iSazonov** [reply]
Try `pwsh -Noprofile.`

**Reply by sergeevabc** [reply]
@iSazonov, no luck again, but a bit of poetry has been born.

Another reboot via pressing reset,
This state of affairs make me so sad.

What else could I try?

**Reply by chrullrich** [reply]
Just to throw out a random idea, can you recover your system if you do this?

1. In one PS window (preferably WPS 5 so it does not kill itself), run `Start-Sleep 10; Stop-Process -Force -Name pwsh`.
2. While it is waiting, start the 7.3 and wait until it is terminated.

If this works to get you the system back without reset, you can try running Procmon to get an idea of what `pwsh` is doing.

**Reply by sergeevabc** [reply]
@chrullrich, done as you said. Powershell 5 terminated 7.3.0 window (not the process) after 5 minutes of struggle. Then I observed pwsh.exe process in Process Hacker window for another 5 minutes with no ability to kill it (task manager app displayed some error saying like it's not possible to kill the existing process, what?!). All these 10 minutes I heard HDD spinning like mad (swapping, perhaps) and saw memory indicator full (100%). Rape it is.

**Reply by chrullrich** [reply]
[That comment went to the wrong place. Repost.]

Ouch. Then the only advice I can give is the usual boilerplate that we are all thoroughly tired of: Try safe mode, turn off, better yet, uninstall, your antivirus software, update your drivers, disconnect all nonessential devices, etc.

**Reply by sergeevabc** [reply]
@chrullrich, I'll stay with 7.2.7 for now then. There is no rush here. I am sure the bug will pop up eventually as we have a large userbase (however not many of us dare to speak publicly, especially of troubles). Thank you for paying attention!

**Reply by kilasuit** [reply]
Windows 7 is not a supported OS for PowerShell 7

**Reply by kilasuit** [reply]
I'm saying you are highly unlikely to get any real support because you are running on an outdated os that technically was never a supported os for PowerShell 7 to run on.

**Reply by sergeevabc** [reply]
Finally, [there is a way](https://github.com/PowerShell/PowerShell/issues/19816#issuecomment-2978689653) to run the latest Pwsh on Windows 7.

``` console
$ $PSVersionTable
Name                           Value
----                           -----
PSVersion                      7.5.1
PSEdition                      Core
GitCommitId                    7.5.1
OS                             Microsoft Windows 6.1.7601 Service Pack 1
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by AleksejEgorov** [comment]
Why did you remove the "Type 'help' to get help" welcome message and docs link? 😿 
They were incredibly user-friendly.

**Reply by iSazonov** [reply]
There was a discussion about "why pwsh creates this header while others (bash) do not".

**Reply by AleksejEgorov** [reply]
Some (like bash) don't, others (like python) do. Still think it was more friendly, especially for first-time users.

**Comment by tekksick** [comment]
Do we have to download all the 24 assets?
All the normal files and a zip file and a tar.gz file? @iSazonov @kilasuit

**Reply by drghughes** [reply]
Just download the file for your system (in your preferred format, if there is more than one).

There are installation instructions in the _Get PowerShell_ section of https://github.com/PowerShell/PowerShell#readme

**Comment by k1f1t** [comment]
how to install PowerShell using command?

**Reply by drghughes** [reply]
Do the installation instruction in the _Get PowerShell_ section of https://github.com/PowerShell/PowerShell#readme help?

**Comment by jltaylor0331** [comment]
Linux user here...just ran the bash script for Evaluation-STIP_Bash.sh in the correct context and received an error: Line 197 Permission denied for powershell/pwsh. Since I'm running in a disconnected virtual environment, I had downloaded the file "Powershell-7.30-linux-x64.tar.gz', renamed it to "powershell.tar.gz", placed it in the Evaluate-STIG directory, and ran Evaluate-STIG_Bash.sh as sudo bash. Then I su - to root and tried it again. No joy.  Is there a permissions problem within the compressed file? Thoughts? Thanks in advance.

**Reply by adityapatwardhan** [reply]
This has been fixed in 7.3.1. Thanks for reporting

**Comment by drghughes** [comment]
Will Microsoft Update update v7.2.7 to v7.3.0?

I've noticed that Microsoft Update has been updating the v7.2.x series a week or so after release, but it's been over a month now, and v7.2.7 is still the installed version.  So do I have to manually update to v7.3.0?

Thanks!

**Reply by kilasuit** [reply]
Microsoft update is always a little behind the times of releases that you could otherwise manually install.

So you can manually update or wait for it to come around though it may have been temp blocked due to some additional issues that the team has found since it's release

**Reply by drghughes** [reply]
Thanks @kilasuit!

I just got v7.2.8 with the automatic updates, so it looks like I'll have to manually install the v7.3.x series if I want it.

![image](https://user-images.githubusercontent.com/20718946/208048325-0631972d-f930-444b-923c-4fb8a219a75c.png)
